syncd/syncd_pfc_watchdog.cpp

```diff
@@ -0,0 +1,260 @@
+#include "syncd_pfc_watchdog.h"
+#include "syncd.h"
+
+#define PFC_WD_POLL_MSECS 50
+
+PfcWatchdog::PfcCounterIds::PfcCounterIds(
+        _In_ sai_object_id_t queue,
+        _In_ sai_object_id_t port,
+        _In_ const std::vector<sai_port_stat_t> &portIds,
+        _In_ const std::vector<sai_queue_stat_t> &queueIds):
+    queueId(queue), portId(port), portCounterIds(portIds), queueCounterIds(queueIds)
+{
+}
+
+void PfcWatchdog::setPortCounterList(
+        _In_ sai_object_id_t queueVid,
+        _In_ sai_object_id_t queueId,
+        _In_ const std::vector<sai_port_stat_t> &counterIds)
+{
+    SWSS_LOG_ENTER();
+
+    PfcWatchdog &wd = getInstance();
+
+    sai_object_id_t portId = queueIdToPortId(queueId);
+    if (portId == SAI_NULL_OBJECT_ID)
+    {
+        return;
+    }
+
+    auto it = wd.m_counterIdsMap.find(queueVid);
+    if (it != wd.m_counterIdsMap.end())
+    {
+        (*it).second->portCounterIds = counterIds;
+        return;
+    }
+
+    auto pfcCounterIds = std::shared_ptr<PfcCounterIds>(new PfcCounterIds(queueId,
```

Comment:
you can use std;;make_shared<> to not use new operator

Reply:
Done